core/invoker/src/main/scala/whisk/core/containerpool/ContainerProxy.scala

```diff
@@ -212,10 +222,34 @@ class ContainerProxy(
     // and we keep it in case we need to destroy it.
     case Event(data: PreWarmedData, _) => stay using data
 
-    // Run was successful
-    case Event(data: WarmedData, _) =>
+    // Init was successful
+    case Event(data: WarmedData, _: PreWarmedData) =>
```

Comment:
Note for myself - In `Started` state following sequence happens

1. proxy transitions to `Running` with `PreWarmedData` (activeCount = 1). 
2. It also then sends msg `WarmedData` (count = 1) in `initializeAndRun` . Assuming this is first `Run` being processed
3. It also then sends msg `WarmedData` with count = 0 

So each call to `initializeAndRun` end up sending 2 `WarmedData` messages. One with count = 0 and one with count at time of start of message

Reply:
for 3 it should send WarmedData with count 0 only to itself for completion notification, and to parent only in case where all active activations are completed. There is a confusing aspect here which is that WarmedData is used BOTH for signaling completion (active count is ignored) on internal state machine AND for signaling "NeedWork" to parent (active count is important).